app/controllers/api/lists_controller.rb

```diff
@@ -21,17 +21,34 @@ def remove
     end
 
     def save
-      ingredients = current_api_user.kitchen_list.ingredients
-      params[:list][:ingredients].each do |ingredient|
+      params_ingredients = params[:list][:ingredients]
+      orig_ingredients = current_api_user.ingredients
+      persisted = []
+
+      #Pull out IDs of 'unpersisted' ingredients coming back from client
+      params_ingredients.each do |ingred|
+        persisted << ingred[:id]
+      end
+
+      #If ID isn't there then delete the persisted ingredient form original ingredients
+      orig_ingredients.each do |orig|
+        if !persisted.include?(orig.id)
+          orig.destroy
+        end
+      end
+
+      #Save new ingredients
+      params_ingredients.each do |ingredient|
         if !ingredient.key?('id')
           Ingredient.create(name: ingredient[:name], kitchen_list_id: 1, category_id: 1)
```

Comment:
@nicholasblock78 , can you remove the hard coded kitchen list id here? Switch it to current_api_user.kitchen_list_id 

Otherwise this will only work for our test user